pages/api/v1/bounty/curator/requests/getAllSentOrReceivedCuratorReq.ts

```diff
@@ -0,0 +1,124 @@
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { isValidNetwork } from '~src/api-utils';
+import { postsByTypeRef } from '~src/api-utils/firestore_refs';
+import { MessageType } from '~src/auth/types';
+import messages from '~src/auth/utils/messages';
+import { getProposalTypeTitle, getSubsquidLikeProposalType, ProposalType } from '~src/global/proposalType';
+import { GET_RECEIVED_CURATOR_REQUESTS, GET_SENT_CURATOR_REQUESTS } from '~src/queries';
+import fetchSubsquid from '~src/util/fetchSubsquid';
+import { BOUNTIES_LISTING_LIMIT } from '~src/global/listingLimit';
+import getEncodedAddress from '~src/util/getEncodedAddress';
+import { getSubSquareContentAndTitle } from 'pages/api/v1/posts/subsqaure/subsquare-content';
+import { EPendingCuratorReqType, IPendingCuratorReq, ISubsquidChildBontyAndBountyRes } from '~src/types';
+import { getDefaultContent } from '~src/util/getDefaultContent';
+import getTokenFromReq from '~src/auth/utils/getTokenFromReq';
+import authServiceInstance from '~src/auth/auth';
+
+const handler: NextApiHandler<{ data: IPendingCuratorReq[]; totalCount: number } | MessageType> = async (req, res) => {
+	storeApiKeyUsage(req);
+
+	const network = String(req.headers['x-network']);
+
+	const { page, userAddress, proposalType, reqType } = req.body;
+
+	try {
+		if (!network || !isValidNetwork(network)) return res.status(400).json({ message: messages.INVALID_NETWORK });
+
+		if (
+			isNaN(page) ||
+			!userAddress?.length ||
+			!getEncodedAddress(userAddress, network) ||
+			![EPendingCuratorReqType.RECEIVED, EPendingCuratorReqType?.SENT].includes(reqType) ||
+			![ProposalType.BOUNTIES, ProposalType.CHILD_BOUNTIES].includes(proposalType || '')
+		) {
+			return res.status(400).json({ message: messages.INVALID_PARAMS });
+		}
+
+		const token = getTokenFromReq(req);
+		if (!token) return res.status(401).json({ message: messages?.INVALID_JWT });
+
+		const user = await authServiceInstance.GetUser(token);
+		if (!user) return res.status(401).json({ message: messages.UNAUTHORISED });
+
+		const encodedUserAddress = getEncodedAddress(userAddress, network);
+
+		const subsquidBountiesRes = await fetchSubsquid({
+			network,
+			query: reqType == EPendingCuratorReqType.RECEIVED ? GET_RECEIVED_CURATOR_REQUESTS : GET_SENT_CURATOR_REQUESTS,
+			variables: {
+				address: encodedUserAddress,
+				limit: BOUNTIES_LISTING_LIMIT,
+				offset: BOUNTIES_LISTING_LIMIT * (page - 1),
+				type_eq: getSubsquidLikeProposalType(proposalType)
+			}
+		});
+
+		if (!subsquidBountiesRes?.data?.proposals?.length) return res.status(400).json({ message: 'No Curator Request Sent!' });
+
+		const subsquidBountiesData = subsquidBountiesRes?.data?.proposals || [];
+
+		const subsquidbountiesIndexes = subsquidBountiesData.map((bounty: { index: number }) => bounty?.index);
+
+		const postDocs = await postsByTypeRef(network, proposalType).where('id', 'in', subsquidbountiesIndexes).get();
```

Comment:
The Firestore query using where('id', 'in', ...) will fail if the array is empty. Add a check to handle this case.